lib/propolis-client/src/instance_spec/builder.rs

```diff
@@ -165,6 +170,44 @@ impl SpecBuilder {
         }
     }
 
+    #[cfg(feature = "falcon")]
+    /// Sets softnpu pci port
+    pub fn set_softnpu_pci_port(
+        &mut self,
+        pci_port: SoftNpuPciPort,
+    ) -> Result<&Self, SpecBuilderError> {
+        self.spec.devices.softnpu_pci_port = Some(pci_port);
```

Comment:
This should also call `self.register_pci_device` so that the builder knows the port is reserved (it will then catch attempts to use it again). Same thing for SoftNpuP9.

Reply:
Done